plugins/rpc/grpc/options.go

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2018 Cisco and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package grpc
+
+import (
+	"github.com/ligato/cn-infra/config"
+	"github.com/ligato/cn-infra/logging"
+	"github.com/ligato/cn-infra/logging/logrus"
+	"github.com/ligato/cn-infra/rpc/grpc"
+	"sync"
+)
+
+var sharedPlugin *Plugin
```

Comment:
What about creating a slice of shared plugins or even a struct:
```golang
type SharedPlugin struct {
   plugins map[string]*Plugin
   sync. RWlock
}
```
Then Sharedplugin() should have a name parameter as input. In this case, in my plugin I do know which plugin I need and I ask for it and for not some shared plugin I have no clue what it does ;) 
In my code I would do:

```
objectstore := SharedPlugin("objectstore")
```
What do you think?

Reply:
A couple of things:

1)  I could see doing it by name instead of {Dep,Config}.  It does raise the question of how to set the Dep, Config of a named shared plugin.  Or what to do if two folks disagree about what that Dep,Config should be for a name.

2)  We could do it with a map/sync.RWlock.  I did it with a sync.Map because it doesn't lock the entire map when you do a LoadOrStore() call.  It was designed for situations where you do mostly reads (which is what we have here).

Reply:
I just realized I haven't pushed the next commit yet, one moment and I'll do so and you can see what I was thinking :)